### DISCUSSION TITLE
raf/polyfill not compatible with edge runtime

### DISCUSSION THREAD

**Discussion by timothymiller** [tamagui/tamagui#1633] [General]
### Current Behavior

Using SSR with `const runtime='edge'` exported on a page throws this error:

```bash
Error: A Node.js API is used (process.hrtime) which is not supported in the Edge Runtime.
```

Removing the `raf/polyfill` import from `_app.tsx` fixes the error, but introduces a styling error (presumably why `raf/pollyfill` is used in the starter)

https://github.com/chrisdickinson/raf

### Expected Behavior

SSR with the next.js `edge` runtime to work as it does with the `node.js` runtime

### Tamagui Version

```markdown
1.58.4
```

### Reproduction

```markdown
Using SSR with `const runtime='edge'` exported on a page throws this error


Error: A Node.js API is used (process.hrtime) which is not supported in the Edge Runtime.
```

**Comment by natew** [comment]
can you try something like

```

// for SSR
if (typeof requestAnimationFrame === 'undefined') {
  globalThis['requestAnimationFrame'] = setImmediate
}
```

or even `= queueMicrotask`

**Comment by timothymiller** [comment]
Replacing `import raf/pollyfill` with
<img width="690" alt="Screenshot 2023-09-01 at 16 43 54" src="https://github.com/tamagui/tamagui/assets/46549361/f537cc4c-f5fe-447f-b0f3-91ff6bbcf0f2">

✅ works for client side only routes.
🔴 It takes the styling away from server side rendered routes:
<img width="659" alt="Screenshot 2023-09-01 at 16 44 29" src="https://github.com/tamagui/tamagui/assets/46549361/932e2e8b-0e9e-47c1-83c4-3e8506f357d6">

**Comment by timothymiller** [comment]
It does shave 1kb off of bundle size on the plus side :)

**Comment by natew** [comment]
Hm let me know what you find, I think this is just a matter of conditionally importing the polyfill depending on the environment, but going to convert to discussions since its a new functionality more than a bug
